sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -650,15 +650,29 @@ public virtual async Task CloseAsync(CancellationToken cancellationToken = defau
         ///   events.
         /// </summary>
         ///
+        /// <param name="events">The events to which the scope will be linked.</param>
+        ///
         /// <returns>The requested <see cref="DiagnosticScope" />.</returns>
         ///
-        private DiagnosticScope CreateDiagnosticScope()
+        private DiagnosticScope CreateDiagnosticScope(IEnumerable<EventData> events)
```

Comment:
Assuming the above comment is where we end up, this could just take a set of the string identifiers and save the work of trying to parse them out.